arangosh/Restore/RestoreFeature.cpp

```diff
@@ -529,6 +529,7 @@ arangodb::Result sendRestoreData(arangodb::httpclient::SimpleHttpClient& httpCli
     arangodb::velocypack::Options opts = arangodb::velocypack::Options::Defaults;
     // do *not* check duplicate attributes here (because that would throw)
     opts.checkAttributeUniqueness = false;
+    opts.validateUtf8Strings = true;
```

Comment:
This is a bit inconsistent: Is it good to do the validation in one of the client tools only? If we do the validation here, shouldn't we also do some validation in arangoimport, which also sends batches of user-input to the server?

But if we validate all input on the server-side anyway, extra validation in the client tools will not provide any benefit. My position is that we need to validate all incoming data on the server-side and that there we need to reject invalid data consistently. Validation in the client tools is not enough, because users can always bypass the client tools and use drivers/APIs directly to send invalid data to the server.

Reply:
Is "validation of all input data on the server-side" something that can be implemented soon/easily?

It would be great if this functionality could also be backported to 3.6.x.
(related ticket: ES-545)

Reply:
I think the changes here are not needed anymore as the server should now reject invalid documents.

Reply:
Ok, shouldn't we then remove that line from RestoreFeature.cpp?

Reply:
It does not do harm to catch the problem earlier here? I think while it is not necessary it saves some traffic.

Reply:
Yes, it will save traffic, but only for the case in which invalid data is imported. And make restoring valid data slower.
That's why usually we don't optimize for the error case.

And it is inconsistent if we only do it in restore. Why don't we do the same client-side validation in arangoimport then?